content/_index.md

```diff
@@ -2,9 +2,39 @@
 title: "🍲 Based Cooking 🍳"
 ---
 
-## Categories
-
-{{< tagcloud >}}
+## What do you want to cook?
+
+<div class="search">
+  <input type="text" id="search" placeholder="Search...">
+  <button class="clear-search">
+    <svg xmlns="http://www.w3.org/2000/svg" class="ionicon" viewBox="0 0 512 512"><title>Backspace</title><path d="M135.19 390.14a28.79 28.79 0 0021.68 9.86h246.26A29 29 0 00432 371.13V140.87A29 29 0 00403.13 112H156.87a28.84 28.84 0 00-21.67 9.84v0L46.33 256l88.86 134.11z" fill="none" stroke="currentColor" stroke-linejoin="round" stroke-width="32"></path><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="32" d="M336.67 192.33L206.66 322.34M336.67 322.34L206.66 192.33M336.67 192.33L206.66 322.34M336.67 322.34L206.66 192.33"></path></svg>
+  </button>
+</div>
+
+<script>
+document.addEventListener('DOMContentLoaded', () => {
+  const rec = document.querySelectorAll('#artlist li')
+  const search = document.querySelector('#search')
+  const clearSearch = document.querySelector('.clear-search')
+
+  search.addEventListener('change', e => {
+    // grab search input value
+    const searchText = e.target.value.toLowerCase()
+    // for each receipe hide all but matched
+    rec.forEach(el => {
+      const recipeName = el.innerText.toLowerCase()
+      const isMatch = recipeName.includes(searchText)
+
+      el.hidden = !isMatch
```

Comment:
```suggestion
      el.classList.toggle('matched-recipe', isMatch && searchText.length !== 0);
```

Adds a CSS class `matched-recipe` to the recipes that match the search text, but the class removed from every recipe if the search box is empty

Reply:
Doing this change exactly as given would not remove the non-matching elements anymore though.

Reply:
Is there anything amiss with just not removing the original line? Seems to work for what I want fine like that.

This actually brings up another issue though. The article list `ul` has responsive CSS that adjusts the number of columns depending on the window size.

Could the `ul`'s CSS too?

Reply:
```suggestion
      el.hidden = !isMatch
      el.classList.toggle('matched-recipe', isMatch && hasFilter);
```

Nope, you can definitely leave that line there just like this.

We could have this code modify the `ul`'s CSS as well, though it probably wouldn't make sense here, since this is inside a loop, and we really only need to set a class for the `ul` once.

I'll redo this suggestion.

Reply:
I'll add that if you press the delete button, it does not remove the highlighted item's class alignment.